node/network/collator-protocol/Cargo.toml

```diff
@@ -30,3 +32,6 @@ sp-core = { git = "https://github.com/paritytech/substrate", branch = "master",
 sp-keyring = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
 polkadot-subsystem-testhelpers = { package = "polkadot-node-subsystem-test-helpers", path = "../../subsystem-test-helpers" }
+
+[features]
+sampling = []
```

Comment:
What's the plan with the feature? We probably don't want to have two binaries for validators depending on whether we want sampling or not. If we want to test this, we can roll it out on some rococo validators to see how they are behaving. And remove the feature altogether.

Reply:
I suggested the feature-flag, so it doesn't go into a release until we've tested it thoroughly on Rococo

Reply:
My suggestion is instead not to merge it until we've tested it thoroughly. The binary can be built from the branch using CI and deployed for testing on Rococo.